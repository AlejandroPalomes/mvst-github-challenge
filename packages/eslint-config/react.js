const { resolve } = require("node:path");

const project = resolve(process.cwd(), "tsconfig.json");

/*
 * This is a custom ESLint configuration for use a library
 * that utilizes React.
 *
 * This config extends the Vercel Engineering Style Guide.
 * For more information, see https://github.com/vercel/style-guide
 *
 */

module.exports = {
  extends: [
    "@vercel/style-guide/eslint/browser",
    "@vercel/style-guide/eslint/typescript",
    "@vercel/style-guide/eslint/react",
  ].map(require.resolve),
  parserOptions: {
    project,
  },
  plugins: ["only-warn"],
  globals: {
    JSX: true,
  },
  settings: {
    "import/resolver": {
      typescript: {
        project,
      },
    },
  },
  ignorePatterns: ["node_modules/", "dist/", ".eslintrc.js", "**/*.css"],
  // add rules configurations here
  rules: {
    "import/no-default-export": "off",
    "@typescript-eslint/explicit-function-return-type": "off",
    "jsx-a11y/click-events-have-key-events": "off",
    "no-console": "off",
    "@typescript-eslint/no-unsafe-member-access": "off",
    "@typescript-eslint/no-explicit-any": "off",
    "react/jsx-no-leaked-render": "off",
    "jsx-a11y/no-static-element-interactions": "off",
    "react/function-component-definition": [2, { "namedComponents": "arrow-function" }]
  },
};
