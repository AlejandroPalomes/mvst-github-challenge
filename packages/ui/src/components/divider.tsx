import { type FC } from "react";

interface DividerProps {
  color?: 700 | 600 | 500 | 400 | 300 | 200 | 100
}

export const Divider: FC<DividerProps> = ({ color = 700}) =>
  <div className={`h-px w-full bg-customGray-${color}`}/>;
