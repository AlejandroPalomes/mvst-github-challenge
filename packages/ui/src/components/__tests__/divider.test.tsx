import { render } from "@testing-library/react";
import { Divider } from "../divider";

describe('Divider test', () => {
  it('should render without crashing', () => {
    const { baseElement } = render(<Divider/>);
    expect(baseElement).toMatchSnapshot();
  });
})