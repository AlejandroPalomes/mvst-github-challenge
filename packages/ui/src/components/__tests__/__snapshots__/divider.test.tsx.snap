// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider test > should render without crashing 1`] = `
<body>
  <div>
    <div
      class="h-px w-full bg-customGray-700"
    />
  </div>
</body>
`;
